src/index.js

```diff
@@ -213,11 +207,24 @@ async function executeActions(params) {
     // a fetchers `resolve` should never throw an exception but report a proper status response.
     // so we consider any exception thrown as application error and propagate it to openwhisk.
     /* istanbul ignore next */
-    log.error(`error while invoking fetchers: ${e}`);
+    log.error(`!!!error while invoking fetchers: ${e}`);
     /* istanbul ignore next */
     return {
       error: `${String(e.stack)}`,
     };
+  } finally {
+    try {
+      // we need to wait for the dummy 404 request, otherwise we have unhandled promise rejections
+      await rejection;
```

Comment:
why is this needed?

Reply:
`fetch404Promise` which we resolve below gets initialized with `rejected`, but might get re-assigned, in which case `rejected` is a dangling promise.

Reply:
as I wrote in helix-chat on slack, I don't think that this is enough. we have to catch the exceptions when they occur, and not eventually.....

Reply:
Really not sure how to move forward here. The change so far seems to fix the issue at hand, but I sense you have a bigger refactor in mind.

Reply:
I did some testing, and I think the following works:

```const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

async function test() {
  console.log('test sleeping...');
  await sleep(100);
  throw Error('overslept.');
}
async function run() {
  const task = test();
  task.catch(() => {
    console.log('error'); // catch the rejection when it's generated
  });
  console.log('run sleeping...');
  await sleep(200);
  const result = await task; // this will throw another rejection.
  return result;
}
run().catch(console.error);
```

try adding the `.catch(() => {})` to the `promise404`